Copy PhD guide into README (#132)
php/phd: README.md

```shell
$ git clone https://github.com/php/phd.git
```

and render your documentation file (`doc-base/.manual.xml` in this example)
in `php` format

```shell
$ php phd/render.php --docbook doc-base/.manual.xml --package PHP --format php
```

## Installation

PhD is distributed via its own PEAR channel, [doc.php.net](http://doc.php.net).
Using this channel is also the easiest way to get it.

### Installation via PEAR

> **_NOTE:_**
> You need a working [PEAR
> installation](https://pear.php.net/manual/en/installation.php).

To install the latest version of PhD:

```shell
$ pear install doc.php.net/phd
```

Verify the installation:
```shell
$ phd --version
PhD Version: 1.0.0-stable
PHP Version: 5.3.3
Copyright(c) 2007-2024 The PHP Documentation Group
```

To see the list of all available PhD packages:

```shell
$ pear remote-list -c doc.php.net
```

Installing the PhD Packages:

```shell
$ pear install doc.php.net/phd_php
Starting to download PhD_PHP-1.0.0.tgz (18,948 bytes)
[...]
install ok: channel://doc.php.net/PhD_PHP-1.0.0

$ pear install doc.php.net/phd_pear
downloading PhD_PEAR-1.0.0.tgz ...
[...]
install ok: channel://doc.php.net/PhD_PEAR-1.0.0

> ```

Reviewer: I would remove this whole block

### Installation from Git

To get the latest and greatest features that have not been released yet,
you can use PhD from Git.

```shell
$ git clone https://github.com/php/phd.git
... output

$ pear install package.xml package_generic.xml package_php.xml package_pear.xml
[...]
install ok: channel://doc.php.net/PhD-1.0.1
install ok: channel://doc.php.net/PhD_Generic-1.0.1
install ok: channel://doc.php.net/PhD_PHP-1.0.1
install ok: channel://doc.php.net/PhD_PEAR-1.0.1
```

To install the standalone Packages:
```shell
for i in package_*.xml; do pear install $i; done
```

```shell
$ phd --version
PhD Version: phd-from-svn
PHP Version: 5.3.3-dev
Copyright(c) 2007-2010 The PHP Documentation Group

> ```

Reviewer: Actually this one as well, as I don't even know if "installation" works, and really should not be done

> ### Using PhD without installation
> 
> You can use PhD without installing it or PEAR by simply cloning the git repository:

Reviewer: ```suggestion
### Using PhD 

To use PhD you must clone the git repo:
```

                epub
                enhancedchm
```

> **_NOTE:_**
> The format packages are provided by separate PEAR packages
> (doc.php.net/PhD_Generic, doc.php.net/PhD_IDE, doc.php.net/PhD_PEAR
> and doc.php.net/PhD_PHP) where only the Generic is installed by
> default.

You can tell by the output of the `--list` option that PhD can also be
used to render the docs as a PDF file or as Unix Man Pages.

To select a format and package, you must use the `-f [formatName]` and
`-P [packageName]` options.

```shell
$ phd -f manpage -P PHP -d .manual.xml
```

This command will output the documentation for PHP functions in the Unix
Man page format.

## Compiling the PhD guide

The PhD guide is this manual you are reading. It lives in
PhD's Git repository under `docs/phd-guide/phd-guide.xml`. If you
installed PhD from Git, you already have it. Otherwise, get it:

```shell
$ svn checkout http://svn.php.net:/repository/phd/trunk/docs/phd-guide
U phd/docs/phd-guide/phd-guide.xml
```

Now you have everything you need. Just type

```shell
$ cd phd/docs/phd-guide/
$ phd -f bigxhtml -d phd-guide.xml
```

There should be an .html file in the directory now. View it with a

> browser!

Reviewer: Possibly removed, or at least the commands need to be updated because we don't use SVN any more.

Author: The `phd-guide.xml` file referenced here is the file that I've copied all this from so I'll just remove this section. :-)

need a `role` attribute describing the type of source code. Examples are
`php`, `html` and `python`.

> **_NOTE:_**
> PhD currently only highlights the code if it is embedded in a `CDATA`
> section.

```xml
<programlisting role="php"><![CDATA[
<?php
echo "Hello world!";
?>
]]></programlisting>
```

By default, PhD uses the source code highlighter that is built into PHP
itself which is only able to highlight PHP code.

If your documentation contains other types of source code or markup,
like XML, HTML, Javascript or any other language, you should try the
[GeSHi](http://qbnz.com/highlighter/) highlighter that is shipped with
PhD:

1.  Install GeSHi from the MediaWiki PEAR channel:

```shell
$ pear channel-discover mediawiki.googlecode.com/svn
$ pear install mediawiki/geshi
```

2.  Use the GeSHi syntax highlighting class when rendering your
    documentation:

```shell
$ phd -g 'phpdotnet\phd\Highlighter_GeSHi' -d phd-guide.xml
```

If you have GeSHi version 1.1.x installed, you should use the
`phpdotnet\phd\Highlighter_GeSHi11x` highlighter, which is adapted to
GeSHi's new API.

Apart from using the highlighter shipped with PhD, you can [build your

> own highlighters](#writing-a-custom-syntax-highlighter).

Reviewer: Does this even still work ?

Author: This doesn't work without PEAR so I think we should remove this too.

# DocBook extensions

PhD has been tailored for PHP and PEAR manuals. To make writing
documentation as easy as possible, some custom tags have been added to the
DTD.

All extensions live in their own XML namespace "`phd:`" which resolves
to <http://www.php.net/ns/phd>. When using one of the attributes or tags
described here, remember to set the namespace:

```xml
xmlns:phd="http://www.php.net/ns/phd"

> ```

Reviewer: Not really relevant anymore.

## General DocBook extensions

The extensions listed here are available in all PhD themes and formats.

### Manual chunking with "phd:chunk" (Attribute)

PhD automatically chooses which sections, chapters or other tags get
their own file (chunk) when using a chunked theme. Sometimes the result
of this automatism is not optimal and you want to fine-tune it. The
attribute "phd:chunk" is offered as a solution by PhD.

#### Allowed values

The value of `phd:chunk` can only be `true` or `false` which force the element
to be chunked or not.

#### Allowed in

`phd:chunk` may be used in every tag that accepts
[db.common.attributes](https://tdg.docbook.org/tdg/5.0/ref-elements.html#common.attributes).

#### Example

```xml
<?xml version="1.0" encoding="utf-8"?>
<preface xmlns="http://docbook.org/ns/docbook"
    xmlns:phd="http://www.php.net/ns/phd"
    xml:id="preface"
    phd:chunk="false"
>
    <info>
    <title>Preface</title>
    ..
    </info>
    ..
</preface>

> ```

Reviewer: We use the standard `annotation` attribute now.

### Generating Table Of Contents: \<phd:toc\> (Tag)

This tag can be used to manually insert a Table Of Contents (TOC)
that creates a list of links to children elements of a tag.

phd:toc-depth

#### Allowed in

`<phd:toc>` can be used everywhere `<para>` elements are allowed.

#### Children

You can add a title with `<title>`.

#### Attributes

| Attribute name | Description | Default value |
|----|----|----|
| phd:element | ID of the element whose children shall be linked | *none* |
| phd:toc-depth | Depth of the TOC/Number of levels | `1` |

> Attributes for \<phd:toc\>

Reviewer: Does this even work? Also conflicts with standard docbook (and really should be a PI if supported)

Author: This is not used in `doc-en`. I'll remove it from the README.

[» some text](https://pear.php.net/package/HTML_QuickForm/docs/latest/HTML_QuickForm/HTML_QuickForm_element.html)

#### Class method links

Class and method name as `phd:linkend` text, separated by a double
colon.

```xml
<phd:pearapi phd:package="HTML_QuickForm" phd:linkend="HTML_QuickForm_element::setName"/>
<phd:pearapi phd:package="HTML_QuickForm" phd:linkend="HTML_QuickForm_element::setName">some text</phd:pearapi>
```

[» HTML_QuickForm_element::setName()](https://pear.php.net/package/HTML_QuickForm/docs/latest/HTML_QuickForm/HTML_QuickForm_element.html#methodsetName)
[» some text](https://pear.php.net/package/HTML_QuickForm/docs/latest/HTML_QuickForm/HTML_QuickForm_element.html#methodsetName)

#### Class variable links

Class and variable name as `phd:linkend` text, separated by a double
colon and a dollar sign before the variable name.

```xml
<phd:pearapi phd:package="Net_Geo" phd:linkend="Net_Geo::$cache_ttl"/>
<phd:pearapi phd:package="Net_Geo" phd:linkend="Net_Geo::$cache_ttl">some text</phd:pearapi>
```

[» Net_Geo::$cache_ttl](https://pear.php.net/package/Net_Geo/docs/latest/apidoc/Net_Geo/Net_Geo.html#var$cache_ttl)

> [» some text](https://pear.php.net/package/Net_Geo/docs/latest/apidoc/Net_Geo/Net_Geo.html#var$cache_ttl)

Reviewer: PEAR is outdated, so not sure if this serves any purpose any more

without changing the internals of PhD.

## Writing a custom syntax highlighter

A syntax highlighter for PhD is nothing more than a simple PHP class
that has two methods: `factory` and `highlight`.

`factory` is static, takes the format name (i.e. `pdf`, `xhtml`,
`troff`) as its only parameter and returns the highlighter instance object
for the given format. The method is called for each output format the
documentation is rendered to.

`highlight` takes three parameters: `text`, `role` and `format`. It is
called whenever a piece of source code needs to be highlighted and
expects the highlighted source code to be returned in the format
of the current rendering format.

Take a look at the provided highlighters, `phpdotnet\phd\Highlighter`,
`phpdotnet\phd\Highlighter_GeSHi` and
`phpdotnet\phd\Highlighter_GeSHi11x`. They will serve as good examples
on how to implement your own highlighter.

Once you wrote your custom source code highlighting class, it's time to
[try it out](#source-code-highlighter).

# Links

PhD 0.1RC1 release announcement:

-   [PhD 0.1RC1
    released](https://bjori.blogspot.com/2007/10/phd-php-based-docbook-renderer-rc1.html)

>     by Hannes Magnusson (PhD 0.1RC1)

Reviewer: I think this can also be removed, nice piece of history but it's 17y old now
